ClosedXML/Excel/XLWorkbook.cs

```diff
@@ -864,21 +864,96 @@ public static Object EvaluateExpr(String expression)
         public XLWorkbook SetLockStructure(Boolean value) { LockStructure = value; return this; }
         public Boolean LockWindows { get; set; }
         public XLWorkbook SetLockWindows(Boolean value) { LockWindows = value; return this; }
-
+        internal HexBinaryValue LockPassword { get; set; }
+        
+        public void Protect(Boolean lockStructure, Boolean lockWindows, String workbookPassword)
+        {
+            if (workbookPassword != null)
+            {
+                var hashPassword = GetPasswordHash(workbookPassword);
+                if (LockPassword != null)
+                {
+                    if (LockPassword != hashPassword)
+                    {
+                        throw new ArgumentException("Invalid password");
+                    }
+                    else
+                    {
+                        if (lockStructure || lockWindows)
+                        {
+                            throw new InvalidOperationException("The workbook is already protected");
+                        }
+                        else
+                        {
+                            //Unprotect workbook using password.
```

Comment:
We are UNprotecting here because `Unprotect` here means to have parameters `lockStructure` and `lockWindows` both FALSE.